include/envoy/common/platform.h

```diff
@@ -180,6 +180,9 @@ struct msghdr {
 
 #if defined(__linux__)
 #include <linux/netfilter_ipv4.h>
+#ifndef IPV6_TRANSPARENT
+#include <linux/in6.h>
```

Comment:
Logically seems good but I'm not sure that if it is good design include the header file if IPV6_TRANSPARENT is not defined.
I think someone should take a look in design perspective.

IMO, following code also makes sense to guarantee that always IPV6_TRANSPARENT is available.
```
#ifndef IPV6_TRANSPARENT
#include <linux/in6.h> // Try to get the IPV6_TRANSPARENT from another header file. In case of debian, it is defined in there.
# ifndef IPV6_TRANSPARENT //Even if include the in6.h but there is no IPV6_TRANSPARENT, let's define it ourselves.
// It is a same value with which defined in linux kernel.
// https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/diff/include/linux/in6.h?id=6c46862280c5f55eda7750391bc65cd7e08c7535
# define IPV6_TRANSPARENT 75
# endif
#endif
```

And I think that it would be great if this comment also be addressed. https://github.com/envoyproxy/envoy/pull/12956#discussion_r482386709 @mattklein123 

FYI, here is a same issue on Ngnix side about 3 years ago. https://trac.nginx.org/nginx/ticket/1487

Reply:
i found this KONG's patch...
https://github.com/Kong/openresty-patches/blob/78e66299b836e254c4794e42aa07599d6d6cb34a/patches/1.13.6.2/nginx-1.13.6_02-listen-transparent.patch#L225-L235

if check defined IPV6_TRANSPARENT only will cause some linux cann't use IPV4_TRANSPARENT even

Reply:
You means that change it like as this? https://github.com/envoyproxy/envoy/issues/12963#issuecomment-686855171

Reply:
i thins most of linux have `<linux/in6.h>`

Reply:
Including `<linux/in6.h>` may fail on some distros (I used internal google one, but I suspect some other distros may be affected as well) due to conflicts with the `<netinet/in.h>`. I did not dig very deep and just added the

```C++
#ifndef IPV6_TRANSPARENT
# define IPV6_TRANSPARENT 75
#endif
```